File: cjxmodel/src/main/java/com/model/cjx/component/MyLoadView.java
package com.model.cjx.component;

import android.app.Activity;
import android.content.Context;
import android.content.res.ColorStateList;
import android.graphics.PorterDuff;
import android.os.Build;
import android.support.v4.content.ContextCompat;
import android.util.AttributeSet;
import android.widget.ProgressBar;

import com.model.cjx.BaseApplication;
import com.model.cjx.R;

/**
 * create by cjx on 2018/12/17
 */
public class MyLoadView extends ProgressBar {
    public MyLoadView(Context context) {
        super(context);
        init();
    }

    public MyLoadView(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    /**
     * 适配android版本的加载view
     */
    private void init() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            if (((Activity) getContext()).getApplication() instanceof BaseApplication) {
                setIndeterminateTintList(ColorStateList.valueOf(BaseApplication.instance.getColorPrimaryDark()));
            } else {
                setIndeterminateTintList(ColorStateList.valueOf(ContextCompat.getColor(getContext(), R.color.cjx_colorPrimaryDark)));
            }

            setIndeterminateTintMode(PorterDuff.Mode.SRC_ATOP);
        } else {
            if (((Activity) getContext()).getApplication() instanceof BaseApplication) {
                setIndeterminateDrawable(BaseApplication.instance.getLoadDrawable());
            } else {
                setIndeterminateDrawable(ContextCompat.getDrawable(getContext(), R.drawable.loading));
            }
        }
    }
}

File: cjxmodel/src/main/java/com/model/cjx/component/MyLoadView.kt
package com.model.cjx.component

import android.annotation.SuppressLint
import android.content.Context
import android.content.res.ColorStateList
import android.graphics.PorterDuff
import android.os.Build
import android.util.AttributeSet
import android.widget.ProgressBar
import com.model.cjx.MyApplication

@SuppressLint("ViewConstructor")
/**
 * Created by cjx on 18-1-9.
 * 旋转进度提示界面
 */
class MyLoadView : ProgressBar {
    constructor(context: Context?) : super(context) {
        val field = ProgressBar::class.java.getDeclaredField("mDuration")
        field.isAccessible = true
        field.setInt(this, 900)
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        init()
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        init()
    }

    /**
     * 适配android版本的加载view
     */
    private fun init() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            indeterminateTintList = ColorStateList.valueOf(MyApplication.instance.getColorPrimaryDark())
            indeterminateTintMode = PorterDuff.Mode.SRC_ATOP
        } else {
            indeterminateDrawable = MyApplication.instance.getLoadDrawable()
        }
    }
}